java/client/src/main/java/com/youtube/vitess/client/cursor/SimpleCursor.java

```diff
@@ -59,10 +59,7 @@ public boolean next() throws SQLException {
   }
 
   @Override
-  protected Row getCurrentRow() throws SQLException {
-    if (row == null) {
-      throw new SQLDataException("no current row");
-    }
+  public Row getCurrentRow() {
```

Comment:
The case I was thinking of is if the user mistakenly calls getCurrentRow() before calling next(). If we don't check for that, then the user will get a generic NullPointerException, which could be confusing. In JDBC's ResultSet class (which I was trying to mimic), they instead check for methods being called in the wrong way, and throw SQLDataException with a message that indicates specifically what the user did wrong.